registry/registry_ui.go

```diff
@@ -330,6 +385,17 @@ func createUpdateNamespace(ctx *gin.Context, isUpdate bool) {
 				ctx.JSON(http.StatusNotFound, gin.H{"error": "Namespace not found. Check the id or if you own the namespace"})
 				return
 			}
+			existingStatus, err := getNamespaceStatusById(id)
+			if err != nil {
+				log.Error("Error checking namespace status: ", err)
+				ctx.JSON(http.StatusInternalServerError, gin.H{"error": "Error checking namespace status"})
+				return
+			}
+			if existingStatus == Approved {
+				log.Errorf("User '%s' is trying to modify approved namespace registration with id=%d", user, id)
+				ctx.JSON(http.StatusForbidden, gin.H{"error": "You don't have permission to modify an approved registration. Please contact your federation administrator"})
```

Comment:
How do we handle key rotations if we don't allow namespace owners to modify their registrations?

Reply:
I think for now we'd better off now allowing them to modify, even the keys, as we aren't dealing with key rotation in the system. #383 . I imagine to do the rotation, we will have a separate endpoint for partial update like `PATCH /registry_ui/namespace/:id/pubkey` so that users know what they are doing. For the full-update like what you pointed out here, we should be strict.